rust/optimized/Cargo.toml

```diff
@@ -9,6 +9,7 @@ name = "countwords"
 path = "main.rs"
 
 [dependencies]
+bumpalo = { version = "3.6.1", features = ["collections"] }
```

Comment:
Ehhhhh... I kind of think this is starting to encroach too much on the "no third party libs" rule.

I did use `fxhash`, but mostly out of convenience and because it's just a hashing function. If pressed, I could have hand-rolled an fnv hasher and it would have done just fine here I think.

But using bumpalo and pushing out a lot of complexity to a third party here. How hard is it to hand-roll a bump allocator specific to this task? Is it as easy as just creating a `Vec<u8>` at the top of `main` and passing that around?

Obviously it's @benhoyt's call here.

(All that aside, this is really cool and I'm impressed with how easy bumpalo makes this sort of optimization.)

Reply:
Err, what if I just copied bumpalo code here? Does that still considered third party?

Reply:
Or maybe we can just use a huge buffer, like `[u8; 1024 * 1024 * 1024]` which is an alternative to bumpalo but that requires a few additional lines of code.